lib/ts/PriorityQueue.h

```diff
@@ -115,6 +117,22 @@ PriorityQueue<T, Comp>::pop()
 
 template <typename T, typename Comp>
 void
+PriorityQueue<T, Comp>::erase(PriorityQueueEntry<T> *entry)
+{
+  if (empty()) {
+    return;
+  }
+
+  _v.remove_index(entry->index);
+
+  // Correct all of the indexes for the removed items
+  for (unsigned int idx = entry->index; idx < _v.length(); idx++) {
```

Comment:
IMO, we can avoid rewriting all of the indexes because this is Binary Heap.
How about this like what we're doing in `PriorityQueue<T, Comp>::pop()` ?

```
_v[entry->index] = _v[_v.length() - 1];
_v.pop();
_bubble_down(entry->index);
```